src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -67,6 +76,58 @@ export type ReactNodeViewConstructor = (
   ...args: Parameters<NodeViewConstructor>
 ) => ReactNodeView;
 
+const REACT_PM_KEY = "reactpmkey";
+
+function findNearestRegistryKey(editorView: EditorView, pos: number) {
+  let parentElement: HTMLElement | null = editorView.domAtPos(pos, 0)
```

Comment:
Can we use `closest(selector)` here with a `[data-react-pm-key]` or something?

Reply:
I don't know if that requires you to actually add things as `data-` attributes instead of in the underlying `dataset`, but it might be nice even if you do have to do that!

Reply:
Looks like `dataset` is a read-write proxy for setting `data-` attributes, so you should be able to use an attribute query to find the closest ancestor using `parentElement.closest("[data-reactpmkey]")`.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Element/closest

Reply:
Oh, right! Awesome, great thought!